pkg/build/cmd/publishgithub.go

```diff
@@ -0,0 +1,134 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/google/go-github/github"
+	"github.com/urfave/cli/v2"
+	"golang.org/x/oauth2"
+)
+
+type repo struct {
+	owner string
+	name  string
+}
+
+type flags struct {
+	create       bool
+	dryRun       bool
+	tag          string
+	repo         *repo
+	artifactPath string
+}
+
+var (
+	errTokenIsEmpty    = errors.New("the environment variable GH_TOKEN must be set")
+	errReleaseNotFound = errors.New(`release not found, use "--create" to create the release`)
+)
+
+func PublishGitHub(ctx *cli.Context) error {
+	gitCtx := context.Background()
+	token := os.Getenv("GH_TOKEN")
+	f := getFlags(ctx)
+	if token == "" {
+		return cli.NewExitError(errTokenIsEmpty, 1)
+	}
+
+	if f.dryRun {
+		return runDryRun(f, token, gitCtx)
+	}
+
+	client := getClient(gitCtx, token)
+	release, res, err := client.Repositories.GetReleaseByTag(gitCtx, f.repo.owner, f.repo.name, f.tag)
+	if err != nil && res.StatusCode != 404 {
+		return cli.NewExitError(err, 1)
+	}
+
+	if release == nil {
+		if f.create {
+			release, _, err = client.Repositories.CreateRelease(gitCtx, f.repo.owner, f.repo.name, &github.RepositoryRelease{TagName: &f.tag})
+			if err != nil {
+				return cli.NewExitError(err, 1)
+			}
+		} else {
+			return cli.NewExitError(errReleaseNotFound, 1)
+		}
+	}
+
+	artifactName := path.Base(f.artifactPath)
+	file, err := os.Open(f.artifactPath)
+	if err != nil {
+		return cli.NewExitError(err, 1)
+	}
+
+	asset, _, err := client.Repositories.UploadReleaseAsset(gitCtx, f.repo.owner, f.repo.name, *release.ID, &github.UploadOptions{Name: artifactName}, file)
+	if err != nil {
+		return cli.NewExitError(err, 1)
+	}
+	fmt.Printf("Asset '%s' uploaded to release '%s' on repository '%s/%s'\nDownload: %s\n", *asset.Name, f.tag, f.repo.owner, f.repo.name, *asset.BrowserDownloadURL)
+	return nil
+}
+
+func getClient(ctx context.Context, token string) *github.Client {
```

Comment:
Can be renamed to `newClient` maybe?

Reply:
Done on https://github.com/grafana/grafana/pull/56349/commits/bdfcaf9265fd4aa2af4e491fbd338698ca88506d